tests/utils.py

```diff
@@ -100,11 +144,25 @@ def setup_test_homeserver(name="test", datastore=None, config=None, reactor=None
                 "database": ":memory:",
                 "cp_min": 1,
                 "cp_max": 1,
+                "check_same_thread": False
             },
         }
 
     db_engine = create_engine(config.database_config)
 
+    # Create the database before we actually try and connect to it, based off
+    # the template database we generate in setupdb()
+    if datastore is None and isinstance(db_engine, PostgresEngine):
+        db_conn = db_engine.module.connect(
+            "dbname=synapse_base user=%s" % (POSTGRES_USER,))
```

Comment:
You can use keyword args here, rather than connection strings: http://initd.org/psycopg/docs/module.html#psycopg2.connect